### PR TITLE
Draft: New bicep markdown documentation

### PR DESCRIPTION
**Pull request by bjompen**
Draft: initial version for testing.

### CODE REVIEW

File: Source/Public/New-BicepMarkdownDocumentation.ps1 [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            foreach ($var in $VariableNames) {
+                $Param = $BuildObject.variables.$var
+                $MDVariables += "| $var | $Param |`n"
+            }
+        }
+$FileDocumentationResult += @"
+
+## Variables
+
+$MDVariables
```
Comment (StefanIvemo) [inline Source/Public/New-BicepMarkdownDocumentation.ps1:118]: Can we move the headers around to have it in the order: parameters, variables, outputs?

Comment (bjompen) [reply Source/Public/New-BicepMarkdownDocumentation.ps1:138]: done and done. We may add this as sortable input in some way as well, but lets see if its necessary first.

**Comment by Kaloszer** [comment]
@bjompen @StefanIvemo 
Is this branch dead? I was looking for something similar for my project. 
I could try to pick this up if this is forgotten?

**Comment by StefanIvemo** [comment]
You are more than welcome if you want to pick it up. Can't remember why we closed it.

**Comment by bjompen** [comment]
I believe it was abandoned due to lack of interest mostly. After the initial request we didn't really hear more, and didn't see the point of working on it.
If you look at the linked issue there are some discussions around this as well, and I believe the maturity of BicepNet is definately worth a revisit. It may be doable to extract the sytaxtree and do this way more useful than this demo implementation..
But go ahead, see if it's doable and usable :)

**Comment by Kaloszer** [comment]
Note, I also use metadata in my bicep files, so if usually you would have something like:

``` bicep
@allowed = [
'northeurope'
]
@description = 'Location of the RG'
param location string = 'northeurope'

...

```

I'm using metadata to add additional information about that particular module, then I have to also add @sys namespace as it doesn't really like it without it.

``` bicep

@sys.allowed = [
'northeurope'
]
@sys.description = 'Location of the RG'
param location string = 'northeurope'

metadata type = 'module'
metadata description = 'Used to deploy a resource group in a defined location'

...

```

I'll need to check out how this works currently, but I suppose I'd need to rebase this branch from main and then see what breaks :) ?

**Pull request merged**